cmd/bosun/sched/template.go

```diff
@@ -323,17 +323,27 @@ func (c *Context) GetMeta(metric, name string, v interface{}) (interface{}, erro
 	return nil, nil
 }
 
-// LeftJoin joins the results of the 2nd and higher expressions onto the results of the first expression.
+// LeftJoin takes slices of results and expressions for which it gets the slices of results.
+// Then it joins the 2nd and higher slice of results onto the first slice of results.
 // Joining is performed by group: a group that includes all tags (with same values) of the first group is a match.
-func (c *Context) LeftJoin(q ...interface{}) (interface{}, error) {
-	if len(q) < 2 {
-		return nil, fmt.Errorf("need at least two expressions, got %v", len(q))
+func (c *Context) LeftJoin(v ...interface{}) (interface{}, error) {
+	if len(v) < 2 {
+		return nil, fmt.Errorf("need at least two values (each can be an expression or result slice), got %v", len(v))
 	}
 	// temporarily store the results in a results[M][Ni] Result matrix:
 	// for M queries, tracks Ni results per each i'th query
-	results := make([][]*expr.Result, len(q))
-	for col, v := range q {
-		queryResults, _, err := c.eval(v, false, false, 0)
+	results := make([][]*expr.Result, len(v))
+	for col, val := range v {
+		var queryResults expr.ResultSlice
+		var err error
+		switch val := val.(type) {
+		case string, *expr.Expr:
+			queryResults, _, err = c.eval(val, false, false, 0)
+		case expr.ResultSlice:
```

Comment:
What if we just added this functionality to the eval function? It's more general, solving this problem for any function that needs to execute expressions.